conditions-dsl.md

```diff
@@ -0,0 +1,147 @@
+The Conditions DSL is a new, universal way of specifying conditional execution
+of CI/CD commands in Semaphore.
+
+Using Conditions you can perform a full or regular expression matches and
+combine them with boolean operations. For example:
+```
+branch == 'master' OR tag =~ '^v1\.'
+```
+
+Currently you can use Conditions to specify when a block should and should not
+run, with more applications coming soon.
+
+- [Formal language definition](#Formal-language-definition)
+- [Usage examples](#usage-examples)
+
+## Formal language definition
+
+Formal language definition in [extended Backus-Naur Form (EBNF)][ebnf] notation:
+
+```
+expression = expression bool_operator term
+           | term
+
+term = "(" expression ")"      
+     | keyword operator string
+     | string operator keyword
+     | string                  
+     | boolean
+
+bool_operator = "and" | "AND" | "or" | "OR"
+
+keyword = "branch" | "BRANCH" | "tag" | "TAG"
+
+operator = "=" | "!=" | "=~" | "!~"
+
+boolean = "true" | "TRUE" | "false" | "FALSE"
+
+string = ? all characters between two single quotes, e.g. 'master' ?
+```           
+
+Each `keyword` in passed expression is replaced with actual value of that
+attribute for current pipeline when expression is evaluated, and then operations
+identified with one of the `operators` from above are executed with those values.
+
+|    KEYWORD     |             ATTRIBUTE IT REPRESENTS              |
+| :------------- | :----------------------------------------------- |
+| branch         | Name of the Git branch from which originated the pipeline that is being executed. |
+| tag            | Name of the Git tag from which originated the pipeline that is being executed. |
+
+
+|  OPERATOR |                 OPERATION RESULT                          |
+| :-------: | :-------------------------------------------------------- |
+|   =       | True if keyword value and given string are equal          |
+|   !=      | True if keyword value and given string are not equal      |
+|   =~      | True if keyword value and given PCRE* string match        |
+|   !~      | True if keyword value and given PCRE* string do not match |
+|   and     | True if expressions on both sides are true                |
+|   or      | True if at least one of two expressions is true           |
+
+\* PCRE = Perl Compatible Regular Expression
+
+
+## Usage examples
+
+### Condition is met always
```

Comment:
You see how "Condition is met" repeats every time. Makes reading harder because when you're scanning to find what interests you, you read only headers.

You can simplify it for example:

- Run always
- Run on any branch
- When branch is master
- When branch starts with “df/”
- Run on any tag
- When branch is master or on any tag

etc

Reply:
First two would have to be `Skip always` and `Skip on any branch` and that is again tied to use case. 

I will change it to this and we can tweak it when more use cases are available.

- Always true
- On any branch
- When branch is master
- When branch starts with “df/”
- On any tag
- When branch is master or on any tag